com.civicrm.hrjobroles/CRM/Hrjobroles/Upgrader.php

```diff
@@ -80,6 +80,66 @@ public function upgrade_1005() {
 
     return TRUE;
   }
+  
+  /**
+   * Deletes cost centre "other" option value or disable it if not in use
+   *
+   * @return bool
+   */
+  public function upgrade_1006() {
+    $jobRoles = civicrm_api3('HrJobRoles', 'get');
+    if ($jobRoles['count'] == 0) {
+      $this->deleteCostCentreOther();
+    }
+    
+    $otherId = $this->retrieveCostCentreOtherId();
+    $inUse = FALSE;
+    $roles = $jobRoles['values'];
+    $pattern = '/\|' . $otherId . '\|/';
+    foreach ($roles as $role) {
+      if (preg_match($pattern, $role['cost_center'])) {
+        $inUse = TRUE;
+        break;
+      }
+    }
+    
+    if (!$inUse) { // disables cost centre other
+      civicrm_api3('OptionValue', 'get', [
+        'option_group_id' => 'cost_centres',
+        'name' => 'Other',
+        'api.OptionValue.create' => ['id' => '$value.id', 'is_active' => 0],
+      ]);
+    }
+    
+    return TRUE;
+  }
+  
+  /**
+   * Deletes cost centre option value with name "other"
+   */
+  private function deleteCostCentreOther() {
+    civicrm_api3('OptionValue', 'get', [
+      'option_group_id' => 'cost_centres',
+      'api.OptionValue.delete' => [
+        'id' => '$value.id',
+        'name' => 'Other'
```

Comment:
Why do you need to get all the option values if you only want to delete one of them? What about doing something like this:

```php
civicrm_api3('OptionValue', 'get', [
   'option_group_id' => 'cost_centres',
   'name' => 'Other',
   'api.OptionValue.delete' => [
      'id' => '$value.id'
   ]
]);
```